src/ngResource/resource.js

```diff
@@ -588,14 +657,23 @@ angular.module('ngResource', ['ng']).
               }
             });
 
+            if (!isInstanceCall && action.cancellable) {
+              var deferred = $q.defer();
+              httpConfig.timeout = deferred.promise;
+              value.$cancelRequest = deferred.resolve;
+            }
+
             if (hasBody) httpConfig.data = data;
             route.setUrlParams(httpConfig,
               extend({}, extractParams(data, action.params || {}), params),
               action.url);
 
-            var promise = $http(httpConfig).then(function(response) {
+            var promise = $http(httpConfig).finally(function() {
+              if (value.$cancelRequest) value.$cancelRequest = angular.noop;
```

Comment:
See https://github.com/angular/angular.js/pull/13210#discussion_r45747318.
We need this in order to enable `deferred` to be gc'ed and we need `$cancelRequest()` to be a function (that does nothing).